src/Catalyst.Core.Modules.Dfs/CoreApi/BlockApi.cs

```diff
@@ -0,0 +1,259 @@
+                    id,
+                    20, // TODO: remove this
+                    cancel: queryCancel.Token,
+                    action: (peer) =>
+                    {
+                        var __ = ProviderFoundAsync(peer, queryCancel.Token).ConfigureAwait(false);
+                    }
+                );
+
+                var got = await bitswapGet;
+                log.Debug("bitswap got the block");
+
+                queryCancel.Cancel(false); // stop the network query.
+                return got;
+            }
+        }
+
+        private async Task ProviderFoundAsync(Peer peer, CancellationToken cancel)
+        {
+            if (cancel.IsCancellationRequested)
+                return;
+
+            log.Debug($"Connecting to provider {peer.Id}");
+            var swarm = await ipfs.SwarmService.ConfigureAwait(false);
+            try
+            {
+                await swarm.ConnectAsync(peer, cancel).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                log.Warn($"Connection to provider {peer.Id} failed, {e.Message}");
+            }
+        }
+
+        public async Task<Cid> PutAsync(byte[] data,
+            string contentType = Cid.DefaultContentType,
+            string multiHash = MultiHash.DefaultAlgorithmName,
+            string encoding = MultiBase.DefaultAlgorithmName,
+            bool pin = false,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            if (data.Length > ipfs.Options.Block.MaxBlockSize)
+                throw new ArgumentOutOfRangeException("data.Length",
+                    $"Block length can not exceed {ipfs.Options.Block.MaxBlockSize}.");
+
+            // Small enough for an inline CID?
+            if (ipfs.Options.Block.AllowInlineCid && data.Length <= ipfs.Options.Block.InlineCidLimit)
+                return new Cid
+                {
+                    ContentType = contentType,
+                    Hash = MultiHash.ComputeHash(data, "identity")
+                };
+
+            // CID V1 encoding defaulting to base32 which is not
+            // the multibase default. 
+            var cid = new Cid
+            {
+                ContentType = contentType,
+                Hash = MultiHash.ComputeHash(data, multiHash)
+            };
+            if (encoding != "base58btc") cid.Encoding = encoding;
+            var block = new DataBlock
+            {
+                DataBytes = data,
+                Id = cid,
+                Size = data.Length
+            };
+            if (await Store.ExistsAsync(cid).ConfigureAwait(false))
+            {
+                log.DebugFormat("Block '{0}' already present", cid);
+            }
+            else
+            {
+                await Store.PutAsync(cid, block, cancel).ConfigureAwait(false);
+                if (ipfs.IsStarted) await ipfs.Dht.ProvideAsync(cid, false, cancel).ConfigureAwait(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)